### PR TITLE
test(menu): add menu API tests

### PR DESCRIPTION
**Pull request by V-Ahmetvaliev**
### Please read and mark the following check list before creating a pull request:

 - [x] I read and followed the [CONTRIBUTING.md](https://github.com/akveo/nebular/blob/master/CONTRIBUTING.md) guide.
 - [x] I read and followed the [New Feature Checklist](https://github.com/akveo/nebular/blob/master/DEV_DOCS.md#new-feature-checklist) guide.
 
 #### Short description of what this resolves:
Tests added for menu API

### CODE REVIEW

File: src/framework/theme/components/menu/menu.spec.ts [hunk 1]
```diff
@@ -3,11 +3,324 @@
...
+      .querySelector('nb-menu:first-child')
+      .querySelectorAll('[nbmenuitem]')
+      .length;
+    const secondMenuListOnInit = twoMenuFixture.nativeElement
+      .querySelector('nb-menu:last-child')
+      .querySelectorAll('[nbmenuitem]')
+      .length;
+    menuService.addItems([addItemObject], 'menuFirst');
+    twoMenuFixture.detectChanges();
+    const firstMenuListItemAdded = twoMenuFixture.nativeElement
```
Comment (yggg) [inline src/framework/theme/components/menu/menu.spec.ts]: `firstMenuItemsCount`?

File: src/framework/theme/components/menu/menu.spec.ts [hunk 2]
```diff
@@ -3,11 +3,324 @@
...
+    const secondMenuListItemAdded = twoMenuFixture.nativeElement
+      .querySelector('nb-menu:last-child')
+      .querySelectorAll('[nbmenuitem]')
+      .length;
+    expect(firstMenuListItemAdded).toEqual(firstMenuListOnInit + 1);
+    expect(secondMenuListItemAdded).toEqual(secondMenuListOnInit);
+  });
+
+  it('addItems() should add new item to DOM', () => {
+    const fixture = createTestComponent([
```
Comment (yggg) [inline src/framework/theme/components/menu/menu.spec.ts]: one line

File: src/framework/theme/components/menu/menu.spec.ts [hunk 3]
```diff
@@ -3,11 +3,324 @@
...
+      {
+        title: 'Existing item',
+      },
+    ]);
+    const addItemObject = {
```
Comment (yggg) [inline src/framework/theme/components/menu/menu.spec.ts]: one line

File: src/framework/theme/components/menu/menu.spec.ts [hunk 4]
```diff
@@ -3,11 +3,324 @@
...
+      title: 'Added item',
+    };
+    const menuListOnInit = fixture.nativeElement.querySelectorAll('li').length;
```
Comment (yggg) [inline src/framework/theme/components/menu/menu.spec.ts]: specificity

File: src/framework/theme/components/menu/menu.spec.ts [hunk 5]
```diff
@@ -3,11 +3,324 @@
...
+    menuService.addItems([addItemObject], 'menu');
+    fixture.detectChanges();
+    const menuListItemAdded = fixture.nativeElement.querySelectorAll('li').length;
+    expect(menuListItemAdded).toBeGreaterThan(menuListOnInit);
+  });
+
+  it('getSelectedItem() should get selected menu item', (done) => {
```
Comment (yggg) [inline src/framework/theme/components/menu/menu.spec.ts]: suite title should start with `sould`

File: src/framework/theme/components/menu/menu.spec.ts [hunk 6]
```diff
@@ -3,11 +3,324 @@
...
+      expect(menuBag.item.title).toEqual(selectedItem.title);
+      done();
+    });
+  }, 1000);
+
+  it('collapseAll() should hide all expanded items', (done) => {
+    const fixture = createTestComponent([
+      {
+        title: 'Menu item collapsed',
+        children: [
```
Comment (yggg) [inline src/framework/theme/components/menu/menu.spec.ts]: one line

File: src/framework/theme/components/menu/menu.spec.ts [hunk 7]
```diff
@@ -3,11 +3,324 @@
...
+import { pairwise, take } from 'rxjs/operators';
+import { NbMenuComponent } from './menu.component';
+
+
+let menuService;
+
+@Component({
+  template: `<nb-menu [items]="items" [tag]="menuTag"></nb-menu>`,
+})
+export class MenuTestComponent {
```
[the remark this answers is not in the document]
Comment (V-Ahmetvaliev) [reply src/framework/theme/components/menu/menu.spec.ts]: Single / Double

File: src/framework/theme/components/menu/menu.spec.ts [hunk 8]
```diff
@@ -3,11 +3,324 @@
...
+            title: 'Child item',
+          },
+        ],
+      },
+    ]);
+    const childList = fixture.nativeElement.querySelector('.menu-item > ul.menu-items');
+    expect(childList.classList).toContain('expanded');
+  });
+
+  it('should expand child menu items', () => {
```
Comment (V-Ahmetvaliev) [inline src/framework/theme/components/menu/menu.spec.ts]: rename

File: src/framework/theme/components/menu/menu.spec.ts [hunk 9]
```diff
@@ -3,11 +3,324 @@
...
+    twoMenuFixture.componentInstance.firstMenuItems = [{ title: 'Home'}];
+    twoMenuFixture.componentInstance.secondMenuItems = [{ title: 'Home'}];
+    twoMenuFixture.componentInstance.firstMenuTag = 'menuFirst';
+    twoMenuFixture.componentInstance.secondMenuTag = 'menuSecond';
+    menuService = twoMenuFixture.componentInstance.menuPublicService;
+    twoMenuFixture.detectChanges();
+    const addItemObject = {
+      title: 'Added item',
+    };
+    const firstMenuListOnInit = twoMenuFixture.nativeElement
```
[the remark this answers is not in the document]
Comment (V-Ahmetvaliev) [reply src/framework/theme/components/menu/menu.spec.ts]: initialFirstMenuItemsCount
afterAddFirstMenuItemsCount

File: src/framework/theme/components/menu/menu.spec.ts [hunk 10]
```diff
@@ -3,11 +3,324 @@
...
+  });
+
+  it('group element should have only span, without link', () => {
+    const fixture = createTestComponent([
+      {
+        title: 'Group item',
+        group: true,
+      },
+    ]);
+    const menuItem = fixture.nativeElement.querySelector('.menu-item');
```
[the remark this answers is not in the document]
Comment (V-Ahmetvaliev) [reply src/framework/theme/components/menu/menu.spec.ts]: changed to .menu-item everywhere

**Pull request merged**